src/h264/h264.h

```diff
@@ -97,9 +139,14 @@ class H264SeqParameterSet
 				frame_crop_bottom_offset	= r.GetExpGolomb();
 			}
 
-			[[maybe_unused]] auto vui_parameters_present_flag = r.Get(1);
-			
-			//TODO: parse vui
+			vui_parameters_present_flag = r.Get(1);
```

Comment:
no need to keep the flag now as the class now has the `std::optional`
```suggestion
			bool vui_parameters_present_flag = r.Get(1);
```

Reply:
Hmm, I see that in most other places in the codebase we do keep the bool `xxxx_present_flag` . My suggestion would be to retain the bool flags as long as it is a field that is referenced by the spec. Wdyt?

Reply:
i want to change the codebase to use this pattern intstead, so we don't have to do

```
 if (vui_parameters_present_flag && vuiParameters) 
```